src/Microsoft.Extensions.Http/DefaultHttpMessageHandlerBuilderPostInitializer.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.Http.Logging;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.Http
+{
+    // Internal so we can change the requirements without breaking changes.
+    internal class DefaultHttpMessageHandlerBuilderPostInitializer : IHttpMessageHandlerBuilderPostInitializer
```

Comment:
open to suggestions about the name of the interface. Intentionally a scary name right now because halloween and I don't expect many people to implement these

Reply:
This feels exactly like an IStartupFilter to me. Why wouldn't we follow that pattern? Is there anything fundamental about it that wouldn't work? Today we have a built in `IStartupFilter` that sets up the DI request scope. Its very similar to this.

Reply:
Would we want to support more than one though? Is this a "real" extension point or just a hack we're doing to ensure we can put stuff in that wraps the pipeline?

And, if we're hard coding this in our core package, maybe we should just go the full distance and always do the logging stuff.

Reply:
What's the suggestion here exactly?

Reply:
> This feels exactly like an IStartupFilter to me. Why wouldn't we follow that pattern? Is there anything fundamental about it that wouldn't work? 

This would work fine. I started with something like that, but I didn't want the complication of having multiple unless we think it's important. 

> Would we want to support more than one though? Is this a "real" extension point or just a hack we're doing to ensure we can put stuff in that wraps the pipeline?

I don't like putting the logging dependency right in the builder. It makes the hairs on the back of my neck stand up. I think if it's useful for others then it's not bad. 

>And, if we're hard coding this in our core package, maybe we should just go the full distance and always do the logging stuff.

I think it's important that it be removable. Some guy will really want that. If you have to replace a lot of default stuff to remove it, then I'd agree with *some guy*.